avro_schema/utils.lua

```diff
@@ -25,7 +25,27 @@ local function copy_fields(from, to, fields)
     end
 end
 
+local function table_find_value(xtable, xvalue)
+    for _, value in pairs(xtable) do
+        if value == xvalue then
+            return true
+        end
+    end
+    return false
+end
+
+-- Analog for `copy_fields` function, except this one copies all fields
+-- except those mentioned in `fields_not_to_copy`.
+local function copy_fields_except(from, to, fields_not_to_copy)
+    for key, value in pairs(from) do
+        if not table_find_value(fields_not_to_copy, key) then
```

Comment:
O(M*N) here can be replaced with O(M+N) using list-to-map for fields_not_to_copy.